sdk/armada-protocol-common/src/common/interfaces/IArmadaManager.ts

```diff
@@ -1,44 +1,99 @@
-import { IAddress, IChainInfo, ITokenAmount } from '@summerfi/sdk-common/common'
+import { ITokenAmount } from '@summerfi/sdk-common/common'
 import { TransactionInfo } from '@summerfi/sdk-common/orders'
 import { IUser } from '@summerfi/sdk-common/user'
+import { IArmadaPool } from './IArmadaPool'
+import { IArmadaPoolId } from './IArmadaPoolId'
+import { IArmadaPoolInfo } from './IArmadaPoolInfo'
+import { IArmadaPosition } from './IArmadaPosition'
+import { IArmadaPositionId } from './IArmadaPositionId'
 
 /**
  * @name IArmadaManager
  * @description Interface for the Armada Protocol Manager which handles generating transactions for a Fleet
  */
 export interface IArmadaManager {
+  /** POOLS */
+
+  /**
+   * @name getPool
+   * @description Get the position of a user in the specified fleet
+   *
+   * @param poolId ID of the pool to retrieve
+   *
+   * @returns IArmadaPool The pool with the specified ID
+   *
+   */
+  getPool(params: { poolId: IArmadaPoolId }): Promise<IArmadaPool>
+
   /**
-   * @name deposit
-   * @description Deposit tokens into the Fleet
+   * @name getPoolInfo
+   * @description Get the extended position information for a position
+   *
+   * @param poolId ID of the pool to retrieve
+   *
+   * @returns IArmadaPoolInfo The extended information of the pool
+   */
+  getPoolInfo(params: { poolId: IArmadaPoolId }): Promise<IArmadaPoolInfo>
+
+  /** POSITIONS */
+
+  /**
+   * @name getPosition
+   * @description Get the position of a user in the specified fleet
+   *
+   * @param positionId ID of the position to retrieve
+   *
+   * @returns IArmadaPosition The position of the user in the fleet
    *
-   * @param chainInfo Chain in which the fleet is
-   * @param fleetAddress Address of the entry point contract for the Fleet
+   */
+  getPosition(params: { positionId: IArmadaPositionId }): Promise<IArmadaPosition>
+
+  /** TRANSACTIONS */
+
+  /**
+   * @name getNewDepositTX
+   * @description Returns the transactions needed to deposit tokens in the Fleet for a new position
+   *
+   * @param poolId ID of the pool to deposit in
    * @param user Address of the user that is trying to deposit
    * @param amount Token amount to be deposited
    *
    * @returns TransactionInfo[] An array of transactions that must be executed for the operation to succeed
    */
-  deposit(params: {
-    chainInfo: IChainInfo
-    fleetAddress: IAddress
+  getNewDepositTX(params: {
```

Comment:
I noticed that the parameter types are defined inline wherever this function (and similar ones) are used. Have you considered creating a separate interface for these parameters? Not sure about theSDK conventions re this

Reply:
Yeah, we have that approach for the constructor of the different classes, but doing the same for each function parameters seems overkilling to me, as those parameters as only used in that specific function. `getNewDepositTX` and `getWithdrawTX` share the same params at the moment, but they are not necessarily related, so using the same interface seems dangerous to me. Let me know if I'm missing your point here